app/assets/javascripts/backbone/views/media_view.coffee

```diff
@@ -1,28 +1,70 @@
-# Handles clicks on media objects and manages the media modal.
+# Handles events on all media browser elements on the page.
+# Controls the media modal (no more than one per page).
 class MS.Views.MediaView extends Backbone.View
 
+  el: 'body'
+
   events:
-    'click a.edit': 'showMediaModal'
-    'click a.new': 'showMediaModal'
+    'click .media-action.edit': 'showMediaModal'
+    'click .media-action.new': 'showMediaModal'
+    'click .media-action.cancel': 'hideMediaModal'
     'click .media-modal .btn-primary': 'submitForm'
     'ajax:complete .media-modal form': 'submitComplete'
+    'click .media-action.delete': 'hideLogModal'
+    'confirm:complete .media-action.delete': 'deleteItem'
+    'ajax:complete .media-action.proceed': 'deleteComplete'
+
+  defineMediaVariables: (link) ->
+    @mediaBox = @$(link).closest('.media-browser')
+    mediaType = @mediaBox.data('media-type')
+    @isLog = mediaType == 'ProjectLog' ? true : false
```

Comment:
Please remember:

1. CoffeeScript doesn't support the `? :` operator.
2. `anything ? true : false` is usually equivalent to just `anything` or in some cases `!!anything`. You should never need to write `? true : false`.